### PR TITLE
haskell: add servant sphinx docs to build

### PR DESCRIPTION
**Pull request by Profpatsch**
Servant builds its user docs from the `doc/` folder with sphinx. This commit adds them to the package.

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ pkgs }:
+{ stdenv, pkgs }:
```
Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: Why not use `pkgs.stdenv`?

Comment (Profpatsch) [inline pkgs/development/haskell-modules/configuration-common.nix]: Oh, didn’t know that exists.

Comment (Profpatsch) [inline pkgs/development/haskell-modules/configuration-common.nix]: On the other hand it’s more explicit that way, right?

Comment (peti) [inline pkgs/development/haskell-modules/configuration-common.nix]: I don't know ... my feeling is that once you expect the entire frickin' package set as an argument, it's kind of silly to add other arguments on top of that. That's why I asked.

Comment (Profpatsch) [inline pkgs/development/haskell-modules/configuration-common.nix]: lol, ok.

Comment (Profpatsch) [inline pkgs/development/haskell-modules/configuration-common.nix]: here you go

**Comment by Profpatsch** [comment]
Next steps will be the cabal docs.

**Pull request merged**